scrapy_poet/injection.py

```diff
@@ -54,6 +56,16 @@ class _UNDEFINED:
     pass
 
 
+class DynamicDeps(UserDict):
```

Comment:
It seems the only thing UserDIct gives now is .data attribute with dict content; if it's not used, subclassing from dict should be fine. This seems to be useful when implementing new methods (which we don't implement), not when using instances: for the user, deps.data would work the same as deps.